dlib/array2d/array2d_kernel.h

```diff
@@ -117,14 +117,24 @@ namespace dlib
         private:
 
             row(T* data_, long cols) : data(data_), nc_(cols) {}
+            row(row&& r)
+            {
+                *this = std::move(r);
+            }
+            row& operator=(row&& r)
+            {
+                std::swap(data, r.data);
+                std::swap(nc_, r.nc_);
+                return *this;
+            }
```

Comment:
```suggestion
            row(row&& r) = default;
            row& operator=(row&& r) = default;
```
This doesn't make the compiler warnings go away too?

Reply:
Hmm, not sure, let's see.